kernel/src/main/java/org/kframework/kprove/KProve.java

```diff
@@ -72,6 +74,14 @@ public int run() {
         Module specModule = compiled._2();
         Rule boundaryPattern = buildBoundaryPattern(compiledDefinition);
 
+        if (kproveOptions.emitJson) {
+            try {
+                files.saveToKompiled("def-module.json", new String(ToJson.apply(compiled._1()), "UTF-8"));
```

Comment:
I have a few issues here:
1. `compiled._1()` returns an entire definition, not only a module, so the name of the file is not really correct.
2. Right now, each time you call `kprove` you may have a different set of claims, so maybe it would be better if you save this in `kprove.temp`. I know we want to eventually optimize this step, so it might not matter then. Thoughts?

Reply:
Hmmmm, saving in `kprove.temp` may be difficult, because I need to access the file afterward. How about I let the user specify the location with `--emit-json-file some_file_name`, and otherwise default to `*-kompiled/kprove-definition.json`?

Reply:
That sounds even more complicated. I'm happy with where it saves right now if we are to fix kprove eventually.

Reply:
Ok, it works for me for now anyway then.

Reply:
But you should still change the name of the file to reflect that you have a definition inside, not just a module.

Reply:
done